searchcore/src/vespa/searchcore/proton/matching/query.cpp

```diff
@@ -59,16 +60,17 @@ inject(Node::UP query, Node::UP to_inject) {
     return query;
 }
 
-std::vector<ProtonLocationTerm *>
+std::vector<LocationTerm *>
 find_location_terms(Node *tree) {
-    std::vector<ProtonLocationTerm *> retval;
+    std::vector<LocationTerm *> retval;
     std::vector<Node *> nodes;
     nodes.push_back(tree);
-    for (size_t i = 0; i < nodes.size(); ++i) {
-        if (auto loc = dynamic_cast<ProtonLocationTerm *>(nodes[i])) {
-            retval.push_back(loc);
+    for (Node * node : nodes) {
```

Comment:
this is just wrong, please revert

Reply:
hmm, did not notice modification of the vector being iterated. Perhaps a unit test for this case is needed.